pkg/engine/params_k8s.go

```diff
@@ -27,8 +27,17 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 		k8sVersion := orchestratorProfile.OrchestratorVersion
 		k8sComponents := api.K8sComponentsByVersionMap[k8sVersion]
 		kubernetesConfig := orchestratorProfile.KubernetesConfig
+		hyperKubeImageBase := kubernetesConfig.KubernetesImageBase
 		kubernetesImageBase := kubernetesConfig.KubernetesImageBase
 
+		if properties.IsAzureStackCloud() {
+			// AzureStack will produce customized HyperKube image.
+			// The hyperKube image and other dependent images are not host on the same repository
+			// So hyperKube image repository is the value of kubernetesConfig.KubernetesImageBase and
+			// other dependent images repository is the value of cloudSpecConfig.KubernetesSpecConfig.KubernetesImageBase
+			kubernetesImageBase = cloudSpecConfig.KubernetesSpecConfig.KubernetesImageBase
```

Comment:
As discussed, we can consider using the `kubernetesConfig.CustomHyperkubeImage` instead